Meziantou.FluentAssertionsAnalyzers/NunitAssertAnalyzerCodeFixProvider.cs

```diff
@@ -494,7 +494,9 @@ private static async Task<Document> Rewrite(Document document, SyntaxNode nodeTo
                         }
                         else if (IsMethod(out expected, hasSymbol, "Count", "EqualTo") || IsMethod(out expected, hasSymbol, "Length", "EqualTo"))
                         {
-                            result = rewrite.UsingShould(arguments[0], "HaveCount", ArgumentList(expected, arguments.Skip(2)));
+                            var argumentTypeSymbol = semanticModel.GetTypeInfo(arguments[0].Expression, cancellationToken).Type;
+                            var replacementMethodName = argumentTypeSymbol.SpecialType == SpecialType.System_String ? "HaveLength" : "HaveCount";

```

Comment:
I'm pretty sure `Type` can be null. Let's be more defensive:

```suggestion
                            var replacementMethodName = argumentTypeSymbol?.SpecialType == SpecialType.System_String ? "HaveLength" : "HaveCount";

```